builtin/logical/transit/path_cache_config.go

```diff
@@ -0,0 +1,124 @@
+package transit
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/vault/helper/keysutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func (b *backend) pathCacheConfig() *framework.Path {
+	return &framework.Path{
+		Pattern: "cache-config",
+		Fields: map[string]*framework.FieldSchema{
+			"cache-type": &framework.FieldSchema{
```

Comment:
Can we default to a preferred cache type instead of erroring out if none is provided?

Reply:
When a new cache is created it will default to `unlimited` (syncmap). Kind of related is whether this parameter should be `required`. Presently, there are only two cache types, one with a size and one without size, so we don't even really need this parameter. However, as soon as we add a 3rd cache-type we will _likely_ need this parameter. The case where we wouldn't need it is if we opt for some logic that only updates the underlying type if it is specified or if it can be inferred (ie there are 3 caches, 2 that do not have a size and one that does -> user specifies a size but not a cache-type, there is only one possibility). In my mind it is slightly simpler in the code and for users to have to specify the cache-type each time they configure it. Thoughts?